great_expectations/expectations/metrics/map_metric.py

```diff
@@ -902,39 +920,37 @@ def _sqlalchemy_column_map_condition_values(
         compute_domain_kwargs, domain_type="identity"
     )
 
-    result_format = metric_value_kwargs["result_format"]
     if "column" not in accessor_domain_kwargs:
         raise ValueError(
             "_sqlalchemy_column_map_condition_values requires a column in accessor_domain_kwargs"
         )
 
-    query = (
-        sa.select(
-            [sa.column(accessor_domain_kwargs.get("column")).label("unexpected_values")]
+    column_name = accessor_domain_kwargs["column"]
+
+    if column_name not in metrics["table.columns"]:
+        raise ge_exceptions.ExecutionEngineError(
+            message=f'Error: The column "{column_name}" in BatchData does not exist.'
         )
+
+    query = (
+        sa.select([sa.column(column_name).label("unexpected_values")])
         .select_from(selectable)
         .where(unexpected_condition)
     )
+
+    result_format = metric_value_kwargs["result_format"]
     if result_format["result_format"] != "COMPLETE":
         query = query.limit(result_format["partial_unexpected_count"])
-    try:
-        return [
-            val.unexpected_values
-            for val in execution_engine.engine.execute(query).fetchall()
-        ]
-    except OperationalError as oe:
-        exception_message: str = "An SQL execution Exception occurred.  "
-        exception_traceback: str = traceback.format_exc()
-        exception_message += (
-            f'{type(oe).__name__}: "{str(oe)}".  Traceback: "{exception_traceback}".'
-        )
-        logger.error(exception_message)
-        raise ge_exceptions.ExecutionEngineError(message=exception_message)
```

Comment:
See above

Reply:
This messes up CLI stdout checks that must be "clean".

Reply:
Shouldn't we still have the `try/except` block and the `raise ge_exceptions.ExecutionEngineError(message=exception_message)`, like you did above? (and just not use logger.error). I worry that removing will make things harder to debug in the future.

Reply:
Oops, just saw your comment below. Just making sure - is this also handled properly in caller?